### PR TITLE
Add login page to on prem

### PR DESCRIPTION
**Pull request by newswangerd**
Adds:
- Login page
- Masthead with user's  name and username
- Authentication checks on every page

Screen Shot 2020-05-11 at 13 39 19-fullpage
Screen Shot 2020-05-11 at 13 38 17-fullpage

https://github.com/ansible/galaxy_ng/issues/75
https://github.com/ansible/galaxy_ng/issues/16

### CODE REVIEW

File: src/api/response-types/user.ts [hunk 1]
```diff
@@ -17,6 +17,16 @@ export class UserAuthType {
...
 }
 
+export class UserAuthType {
+  id: number;
+  username: string;
+  first_name?: string;
+  last_name?: string;
+  email?: string;
+  groups: { id: number; name: string }[];
+  is_partner_engineer: boolean;
```
Comment (ironfroggy) [inline src/api/response-types/user.ts:27]: Do on-premise instances _have_ "partner engineer" users?

File: src/containers/login/login.tsx [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+
+  private onLoginButtonClick = event => {
+    ActiveUserAPI.login(this.state.usernameValue, this.state.passwordValue)
+      .then(result => {
+        ActiveUserAPI.getUser().then(() =>
+          this.setState({ redirect: this.redirectPage }),
+        );
+      })
+      .catch(result =>
+        this.setState({ passwordValue: '', showHelperText: true }),
```
Comment (ironfroggy) [inline src/containers/login/login.tsx:87]: Is there any other kind of error that could come back than an incorrect password?

Invalid user?
Deactivated account?
500 Internal server error?

File: src/api/response-types/user.ts (hunk 1 above)
Comment (newswangerd) [reply src/api/response-types/user.ts:27]: This is for backwards compatibility. The API in cloud.redhat.com returns an `is_partner_engineer` flag and doesn't have a `groups` attribute yet. We'll probably rename this to `admin` or something like that once we can do so without messing up cloud.

File: src/containers/login/login.tsx (hunk 2 above)
Comment (newswangerd) [reply src/containers/login/login.tsx:87]: Internal error is a good idea, I'll add that.

My thought for not returning an error like `invalid user` or `deactivated account` is that this gives a potential attacker less information about what user accounts are available to break into. I don't want an attacker to be able to determine if there is an `admin` or `root` user just by putting in the username with a blank user.

Comment (newswangerd) [reply src/containers/login/login.tsx:87]: Though now that I think about it, an attacker could just use the api...

**Pull request merged**